src/inmanta/server/services/orchestrationservice.py

```diff
@@ -468,117 +395,155 @@ async def delete_version(self, env: data.Environment, version_id: int) -> Apiret
     async def reserve_version(self, env: data.Environment) -> int:
         return await env.get_next_version()
 
-    async def _put_version(
+    def _create_dao_resources_from_api_resources(
         self,
-        env: data.Environment,
-        version: int,
+        env_id: uuid.UUID,
         resources: List[JsonType],
         resource_state: Dict[ResourceIdStr, Literal[ResourceState.available, ResourceState.undefined]],
-        unknowns: List[data.UnknownParameter],
-        version_info: Optional[JsonType] = None,
-        resource_sets: Optional[Dict[ResourceIdStr, Optional[str]]] = None,
-        partial_base_version: Optional[int] = None,
-        *,
-        connection: asyncpg.connection.Connection,
-    ) -> None:
+        resource_sets: Dict[ResourceIdStr, Optional[str]],
+        set_version: Optional[int] = None,
+    ) -> Dict[ResourceIdStr, data.Resource]:
         """
-        :param resources: a list of serialized resources
+        Convert the resources sent to the put_version or put_partial endpoint to dao Resource objects.
```

Comment:
Could you elaborate a bit on what this means exactly? If I understand correctly, in addition to plainly creating the DAO, it updates the version field in the resource's attributes (if specified), verifies that the requires graph is closed within the provided resources, and sets up the `provides` edge for all cross agent dependencies. The end result are fully configured resource objects, including set up of inter-resource fields.